src/NuGet.Clients/PackageManagement.VisualStudio/Telemetry/NuGetProjectTelemetryService.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Common;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+
+namespace NuGet.PackageManagement.Telemetry
+{
+    public class NuGetProjectTelemetryService
+    {
+        public static NuGetProjectTelemetryService Instance = new NuGetProjectTelemetryService(
+            NuGetTelemetryService.Instance);
+
+        private readonly NuGetTelemetryService _nuGetTelemetryService;
+        private readonly string _nuGetVersion;
+
+        public NuGetProjectTelemetryService(NuGetTelemetryService telemetryService)
```

Comment:
I personally prefer not to add null checks in "internal" code paths. It's a lot of extra code for little debugging benefit. APIs that are meant for external consumption are another story.  Upon inspection of our codebase, I have observed a mixture of approaches (non-null check validation, null check validation, no validation at all). Until we have a comprehensive strategy, I'll ere on the defensive side as you requested.